src/net_processing.cpp

```diff
@@ -1218,6 +1220,10 @@ void PeerManagerImpl::FinalizeNode(const CNode& node)
     }
     WITH_LOCK(g_cs_orphans, m_orphanage.EraseForPeer(nodeid));
     m_txrequest.DisconnectedPeer(nodeid);
+    // Do not check whether peer is registered for reconciliation here, but rather delegate checks
+    // to the module. Otherwise it's easy to skip deleting an intermediate state (e.g., we store
+    // salt for peers before we register them for reconciliation).
```

Comment:
This comment seems unnecessary. We're simply telling the `TxReconciliationTracker` to clear all state for this peer. That seems straightforward enough to not require a comment (similarly for the orphanage and txrequest modules).